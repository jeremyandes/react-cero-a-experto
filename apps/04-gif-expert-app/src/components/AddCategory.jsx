import { useState } from "react"
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {
    const [inputValue, setInputValue] = useState('Jamal Murray');

    const onInputChange = ({ target }) => {
        setInputValue(target.value);
    }

    const onFormSubmit = (event) => {
        event.preventDefault();
        if (inputValue.trim().length > 1) {
            onNewCategory(inputValue);
            setInputValue('');
        }
    }

    return (
        <form onSubmit={onFormSubmit}>
            <input
                type='text'
                placeholder='Buscar gifs'
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}
