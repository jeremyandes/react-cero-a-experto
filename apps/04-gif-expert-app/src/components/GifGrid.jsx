const baseUrl = 'https://api.giphy.com/v1/gifs/search';
const getGifs = async (category) => {
    const url = `${baseUrl}?api_key=h0lXkO2rN89vY3PLpQp23Se8J0su2WsP&q=${category}&limit=10`;
    const resp = await fetch(url);
    const { data = [] } = await resp.json();

    const gifs = data.map(image => ({
        id: image.id,
        title: image.title,
        url: image.images.downsized_medium.url
    }))
    console.log(gifs);
}

export const GifGrid = ({ category }) => {
    getGifs(category);

    return (
        <>
            <h3>{category}</h3>
        </>
    )
}
