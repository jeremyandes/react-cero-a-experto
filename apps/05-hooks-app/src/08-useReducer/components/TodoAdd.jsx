import { useEffect } from "react";
import { useForm } from "../../hooks/useForm";

export const TodoAdd = ({ onNewTodo }) => {
    const { description, onInputChange, onResetForm } = useForm({
        description: ''
    })

    const onFormSubmit = (event) => {
        event.preventDefault();
        if (description.length === 0) return;

        const newTodo = {
            id: new Date().getTime(),
            done: false,
            description,
        }

        onNewTodo(newTodo);
        onResetForm();
    }

    return (
        <>
            <form onSubmit={onFormSubmit}>
                <input type="text" name="description" value={description} onChange={onInputChange} placeholder="QuÃ© hay que hacer?" className="form-control" />
                <button className="btn btn-dark mx-0 mt-3" type="submit">Agregar</button>
            </form>
        </>
    )
}