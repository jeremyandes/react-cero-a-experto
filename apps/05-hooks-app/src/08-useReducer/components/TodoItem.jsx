export const TodoItem = ({ item }) => {
    return (
        <li key={item.id} className="list-group-item d-flex justify-content-between align-items-center">
            <h5>{item.description}</h5>
            <button className="btn btn-danger">Borrar</button>
        </li>
    )
}