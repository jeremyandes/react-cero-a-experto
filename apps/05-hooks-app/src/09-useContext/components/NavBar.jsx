import { Link } from "react-router-dom"

export const NavBar = () => {
    return (
        <>
            <Link className='me-3' to="/login">Login</Link>
            <Link className='me-3' to="/">Home</Link>
            <Link className='me-3' to="/about">About</Link>
        </>
    )
}
