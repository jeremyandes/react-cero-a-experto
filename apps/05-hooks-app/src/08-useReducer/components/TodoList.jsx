import { TodoItem } from "./TodoItem"

export const TodoList = ({ todos = [], onDeleteTodo }) => {
    return (
        <>
            <ul className="list-group">
                {
                    todos.map(todo => (
                        <TodoItem key={todo.id} item={todo} onDeleteTodo={onDeleteTodo} />
                    ))
                }
            </ul>
        </>
    )
}