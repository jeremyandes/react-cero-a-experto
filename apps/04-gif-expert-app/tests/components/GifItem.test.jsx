import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Testing <GifItem />', () => {

    const title = 'Michael Jordan';
    const url = 'https://media0.giphy.com/media/l3q2JCu9lep6dAmyY/giphy.gif?cid=4f32db46skaam7bco6t1ev32fy7dvl6l2227o7vualbk7pik&rid=giphy.gif&ct=g'

    test('should match with snapshot', () => {
        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    })

    test('should show image with URL and ALT attribute', () => {
        render(<GifItem title={title} url={url} />);
        // screen.debug();
        // expect(screen.getByRole('img').src).toBe(url);
        // expect(screen.getByRole('img').alt).toBe(title);

        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    })

    test('should show title in component', () => {
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
    })


})