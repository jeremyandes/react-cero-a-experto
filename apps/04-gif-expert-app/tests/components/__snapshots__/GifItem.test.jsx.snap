// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Michael Jordan"
      src="https://media0.giphy.com/media/l3q2JCu9lep6dAmyY/giphy.gif?cid=4f32db46skaam7bco6t1ev32fy7dvl6l2227o7vualbk7pik&rid=giphy.gif&ct=g"
    />
    <p>
      Michael Jordan
    </p>
  </div>
</div>
`;
